framework/mesh/mesh_matrix3x3.h

```diff
@@ -68,219 +57,177 @@ struct Matrix3x3
     return R;
   }
 
-  /// Copy constructor
-  Matrix3x3& operator=(const Matrix3x3& inM)
+  /// Matrix addition operator
+  Matrix3x3 operator+(const Matrix3x3& other) const
   {
-    for (int k = 0; k < 9; ++k)
-      this->vals[k] = inM.vals[k];
-    return *this;
+    Matrix3x3 result;
+    std::transform(
+      vals.begin(), vals.end(), other.vals.begin(), result.vals.begin(), std::plus<>());
+    return result;
   }
 
-  /// Matrix addition operator.
-  Matrix3x3 operator+(const Matrix3x3& inM)
+  /// Matrix subtraction operator
+  Matrix3x3 operator-(const Matrix3x3& other) const
   {
-    Matrix3x3 oM;
-    for (int k = 0; k < 9; ++k)
-      oM.vals[k] = this->vals[k] + inM.vals[k];
-    return oM;
+    Matrix3x3 result;
+    std::transform(
+      vals.begin(), vals.end(), other.vals.begin(), result.vals.begin(), std::minus<>());
+    return result;
   }
 
-  /// Matrix subtraction operator.
-  Matrix3x3 operator-(const Matrix3x3& inM)
+  /// Matrix scalar multiplication operator
+  Matrix3x3 operator*(const double scalar) const
   {
-    Matrix3x3 oM;
-    for (int k = 0; k < 9; ++k)
-      oM.vals[k] = this->vals[k] - inM.vals[k];
-    return oM;
+    Matrix3x3 result;
+    std::transform(
+      vals.begin(), vals.end(), result.vals.begin(), [scalar](double val) { return val * scalar; });
+    return result;
   }
 
-  /// Matrix multiply with scalar.
-  Matrix3x3 operator*(const double value)
+  /// In-place scalar multiplication operator
+  Matrix3x3& operator*=(const double scalar)
   {
-    Matrix3x3 oM;
-    for (int k = 0; k < 9; ++k)
-    {
-      oM.vals[k] = this->vals[k] * value;
-    }
-    return oM;
+    for (auto& val : vals)
+      val *= scalar;
+    return *this;
   }
 
   /// Matrix multiply with vector.
   Vector3 operator*(const Vector3& vec) const
   {
-    double i_vec[] = {vec.x, vec.y, vec.z};
-    double o_vec[] = {0.0, 0.0, 0.0};
+    std::array<double, 3> i_vec = {vec.x, vec.y, vec.z}; // Input vector as std::array
+    std::array<double, 3> o_vec = {0.0, 0.0, 0.0};       // Output vector initialized to 0
 
     for (int i = 0; i < 3; ++i)
-    {
       for (int j = 0; j < 3; ++j)
-      {
         o_vec[i] += this->GetIJ(i, j) * i_vec[j];
-      }
-    }
-    Vector3 oV(o_vec[0], o_vec[1], o_vec[2]);
-    return oV;
+
+    return Vector3(o_vec[0], o_vec[1], o_vec[2]); // Return as a Vector3
```

Comment:
I think keeping Vector3 is clearer to read (to me). Previously Vector3 was created on a separate line and then returned on the next, I just combined the 2 lines. Unresolve this one if i am incorrect.

Reply:
I've seen clang-tidy suggest to use brackets on things like this many times in the past, but we can keep it. My thought is that since the return type is `Vector3` it should be clear that the return with `{...}` is the constructor argument list.